src/System.Threading.Timer/tests/TimerFiringTests.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using Xunit;
+
+public class TimerFiringTests
+{
+    [Fact]
+    public void Timer_Fires_After_DueTime_Ellapses()
+    {
+        AutoResetEvent are = new AutoResetEvent(false);
+
+        using (var t = new Timer(new TimerCallback((object s) =>
+        {
+            are.Set();
+        }), null, TimeSpan.FromMilliseconds(250), TimeSpan.FromMilliseconds(Timeout.Infinite) /* not relevant */))
+        {
+            Assert.True(are.WaitOne(TimeSpan.FromMilliseconds(750)));
+        }
+    }
+
+    [Fact]
+    public void Timer_Fires_AndPassesStateThroughCallback()
+    {
+        using (var t = new Timer(new TimerCallback((object s) =>
+        {
+            Assert.NotNull(s);
+        }), new object(), TimeSpan.FromMilliseconds(100), TimeSpan.FromMilliseconds(Timeout.Infinite) /* not relevant */)) { }
+    }
+
+    [Fact]
+    public void Timer_Fires_AndPassesNullStateThroughCallback()
+    {
+        using (var t = new Timer(new TimerCallback((object s) =>
+        {
+            Assert.Null(s);
+        }), null, TimeSpan.FromMilliseconds(100), TimeSpan.FromMilliseconds(Timeout.Infinite) /* not relevant */)) { }
+    }
+
+    [Fact]
+    public void Timer_Fires_After_DueTime_AndOn_Period()
+    {
+        int count = 0;
+        AutoResetEvent are = new AutoResetEvent(false);
+
+        using (var t = new Timer(new TimerCallback((object s) =>
+        {
+            count++;
+            if (count >= 2)
+            {
+                are.Set();
+            }
+        }), null, TimeSpan.FromMilliseconds(500), TimeSpan.FromMilliseconds(50)))
```

Comment:
Can you use a smaller value than half a second?  I realize it's not much, but with lots of tests, these values add up and make the tests take longer.